types/src/auction.rs

```diff
@@ -456,140 +458,66 @@ pub trait Auction:
-                .ok_or(Error::MissingKey)?
-                .into_uref()
-                .ok_or(Error::InvalidKeyVariant)?;
-            let tmp_delegator_reward_purse = self
-                .mint(total_delegator_payout)
-                .map_err(|_| Error::MintReward)?;
-            self.transfer_purse_to_purse(
-                tmp_delegator_reward_purse,
-                delegator_reward_purse,
-                total_delegator_payout,
-            )
-            .map_err(|_| Error::DelegatorRewardTransfer)?;
-        }
-
-        self.record_era_info(era_id, era_info)?;
-
-        Ok(())
-    }
-
-    /// Allows delegators to withdraw the seigniorage rewards they have earned.
-    /// Pays out the entire accumulated amount to the destination purse.
-    fn withdraw_delegator_reward(
-        &mut self,
-        validator_public_key: PublicKey,
-        delegator_public_key: PublicKey,
-    ) -> Result<U512> {
-        let account_hash = AccountHash::from_public_key(&delegator_public_key, |x| self.blake2b(x));
-        if self.get_caller() != account_hash {
-            return Err(Error::InvalidPublicKey);
-        }
-
-        let mut bids = detail::get_bids(self)?;
-
-        let bid = match bids.get_mut(&validator_public_key) {
-            Some(bid) => bid,
-            None => return Err(Error::ValidatorNotFound),
-        };
-
-        let delegator = match bid.delegators_mut().get_mut(&delegator_public_key) {
-            Some(delegator) => delegator,
-            None => return Err(Error::DelegatorNotFound),
-        };
-
-        let reward_amount = *delegator.reward();
 
-        if reward_amount.is_zero() {
-            return Ok(reward_amount);
-        }
-
-        let source_purse = self
-            .get_key(DELEGATOR_REWARD_PURSE_KEY)
-            .ok_or(Error::MissingKey)?
-            .into_uref()
-            .ok_or(Error::InvalidKeyVariant)?;
-
-        self.transfer_purse_to_account(source_purse, account_hash, reward_amount)
-            .map_err(|_| Error::WithdrawDelegatorReward)?;
-
-        delegator.zero_reward();
-
-        detail::set_bids(self, bids)?;
-
-        Ok(reward_amount)
-    }
-
-    /// Allows validators to withdraw the seigniorage rewards they have earned.
-    /// Pays out the entire accumulated amount to the destination purse.
-    fn withdraw_validator_reward(&mut self, validator_public_key: PublicKey) -> Result<U512> {
-        let account_hash = AccountHash::from_public_key(&validator_public_key, |x| self.blake2b(x));
-        if self.get_caller() != account_hash {
-            return Err(Error::InvalidPublicKey);
-        }
-
-        let mut bids = detail::get_bids(self)?;
-
-        let bid = match bids.get_mut(&validator_public_key) {
-            Some(bid) => bid,
-            None => return Err(Error::ValidatorNotFound),
-        };
+            // Update stakes for a validator that has not been slashed
+            if let Some(bid) = bids.get_mut(&public_key) {
```

Comment:
I figured since this condition is present in both update_* functions (originally it was also reading from global state on each call), so why not simplify them and avoid extra branching there

Reply:
I would prefer that we keep it as-is and continue to do this check here.

Reply:
Sure, I reverted the logic as much as I could incl. this check